src/components/elements/button/button.styles.ts

```diff
@@ -0,0 +1,107 @@
+'use client';
+
+import { keyframes } from 'styled-components';
+
+import type { ButtonProps } from './button.types';
+
+import { css, styled } from '@/styles';
+
+export const rotate = keyframes`
+    from {
+      transform: rotate(0deg);
+    }
+
+    to {
+      transform: rotate(360deg);
+    }
+`;
+
+export const StyledButton = styled.button<ButtonProps>`
+  position: relative;
+  z-index: 1;
+
+  font-size: ${({ theme, small }) => theme.fontSizes.button[small ? 'small' : 'default']};
+  font-weight: ${({ theme }) => theme.fontWeight.medium};
+
+  padding: ${small => (small ? '20px 24px' : '16px 24px')};
+
+  border-radius: 12px;
+  display: inline-flex;
+  gap: 8px;
+  align-items: center;
+
+  ${props => {
+    switch (props.variant) {
+      case 'primary':
```

Comment:
For the primary version, it seems there are two variants with different hover/focus styles. I think it is not included here.